Har 2512 useamman urakan valinta tilannekuvassa

Kauko opened the pull request:
1. Urakat-palvelun  "hae käyttäjän urakat aikavälillä" palauttaa nyt mäpin, jossa on mukana myös hy tiedot
2. Tilannekuvaan lisätty palvelu urakoiden hakemiselle
3. Tilannekuvan tiedot-ns:ssä käsitellään urakoita suodattimissa

Kauko committed:
hae-kayttajan-urakat-aikavalilta tarkastaa oikeudet ja palauttaa hy tiedot

Aiemmin funktio ei tarkastanut oikeuksia, vaan luotti sokeasti
pyytäjään. Funktiolle on ollut aina mahdollista syöttää valmiiksi
tiedossa oleva urakka-id, jolloin funktio vaan palauttaa sen suoraan.
Nyt ennen palautusta tarkastetaan, että syötetty urakka-id on varmasti
sellainen, johon käyttäjällä on oikeudet.

Funktio myös palauttaa nyt urakoiden tiedot vektorissa mäppejä.
Yksi mäppi kuvastaa hallintayksikköä, jossa urakat ovat vektorissa.

Kauko committed:
Lisää kyselyihin urakat-aikavalilta funktion haluamat kentät

Kauko committed:
Ota tilannekuvassa ja ilmoituksissa käyttöön muutettu urakat-funktio

Kauko committed:
Mahdollisuus hakea käyttäjän urakat monesta hallintayksiköstä

Kauko committed:
Lisää tilannekuvaan palvelu urakoiden hakemiselle

Kauko committed:
Reaction<! voi antaa alkuarvon

Kauko committed:
Hae käyttäjälle urakat tilannekuvan tietoihin

Kauko committed:
Prototyyppi käyttöliittymästä

Kauko committed:
Lähetä palvelinpyynnössä valittujen urakoiden id:t

Kauko committed:
Valitse urakat pudotusvalikoista

Kauko committed:
Piilota murupolku tilannekuvassa

Kauko committed:
Resetoi valitut urakat tilannekuvaan tultaessa

Kauko committed:
Älä näytä pudotusvalikkoa jos se on tyhjä

Kauko committed:
Korjaa aluesuodattimien yhdistäminen ja urakoiden valintalogiikka

Pelkkä merge-with merge ei toimi, koska emme halua käyttää
suodattimina urakoita joita ei löydy uudesta setistä. Täytyy siis
käyttää ainoastaan urakoita jotka löytyvät uudesta setistä, mutta
jos nämä olivat myös vanhassa, käytetään vanhan arvoa.

Samalla lisättiin ominaisuus, jolla uusi urakka valitaan, jos se
kuuluu hallintayksikköön, josta käyttäjä on valinnut kaikki urakat.
Samoin urakkaa ei ikinä valita, jos se kuuluu hallintayksikköön,
josta käyttäjä ei ole valinnut yhtään urakkaa.

Jos hallintayksikössä ei ole urakoita, tämä logiikka ei päde niihin.
Näin voi käydä esim historiakuvassa.

Kauko committed:
Pidä aluesuodattimet suoraan osana suodattimet-atomia

Aivopierun korjausta, aikaisemmassa versiossa oli bugi,
jossa käyttäjän tekemät muutokset valittuihin suodattimiin
eivät ikinä olleet osana (nyt poistettua) aluesuodattimet-atomia.

Kauko committed:
Lisää mahdollisuus lyhentää urakoiden nimiä

Urakoiden nimiä lähdetään lyhentämään siten, että aluksi
koitetaan lyhentää tiettyjä sanoja, esim urakka -> ur. Jos nämä
keinot eivät auta, katkaistaan nimi kahtia siten, että keskellä
olevat kirjaimet korvataan kahdella pisteellä ..

Kauko committed:
Muuta tilannekuvatest käyttämään uutta urakanhakutapaa

Kauko committed:
Älä valitse urakkaa automaattisesti, kun murupolun tila on "koko maa"

Jarzka commented on src/cljc/harja/fmt.cljc:
```diff
@@ -50,6 +50,103 @@
+            ;; clj puolella käyttäen reader conditionaleja, mutta
+            ;; samapa se on toistaa kaikki näin.
+            (re-find #"alueurakka" nimi)
+            (str/replace nimi #"alueurakka" "ur.")
```
Eikö tässä voisi laittaa nimen lowercaseksi ja tehdä vertailun sitten?

Kauko replied:
Miten sitten korvaat alkuperäisestä stringistä esim "Alueurakka" sanan? Ei voi kuitenkaan koko stringiä muuttaa lowercaseksi.

Jarzka replied:
Niin joo, siinä tapauksessa ei taida toimia

Jarzka commented on src/cljc/harja/fmt.cljc:
```diff
@@ -50,6 +50,103 @@
+  kirjaimet kahdella pisteellä .."
+  ([nimi] (lyhennetty-urakan-nimi nimi urakan-nimen-oletuspituus))
+  ([nimi pituus]
+    (loop [nimi nimi]
```
Toimisiko tämä ilman looppia käyttämällä cond-> ?

Kauko replied:
Ajatuksena oli, että tehdään yksi muokkaus kerrallaan, ja tarkastetaan sen jälkeen pituus uusiksi. cond-> toimisi varmaan myös, jos jokainen predikaatti sisältäisi andin ja pituustarkastuksen. Ihan hyvä pointti!

En tiedä onko hyötyä lähteä muuttamaan..

Jarzka commented on src/cljs/harja/tiedot/tilannekuva/tilannekuva.cljs:
```diff
@@ -11,14 +12,20 @@
 
+(def
+  ^{:doc   "Kuinka pitkä urakan nimi hyväksytään pudotusvalikkoon"
+   :const true}
```
Eikö def ole aina const?

Kauko replied:
Kuulemma ei. :const true tarkoittaa, että kääntäjä voi optimoida tän, kun tietää että arvo ei ikinä muutu.

Kauko committed:
Älä näytä tilannekuvassa organisaatiorajoja

Kauko committed:
Palauta urakoiden mukana niiden alueet

Kauko committed:
pohjois-pohjanmaa -> pohjois-pohjanmaa ja kainuu testidatassa

Kauko committed:
Piirrä valitut urakat kartalle

Kauko committed:
Merge branch 'develop' into HAR-2512_useamman_urakan_valinta_tilannekuvassa

Kauko committed:
Lisää tilannekuvan urakoille pakollinen :type

Kauko committed:
Siirrä urakan :type avain oikeaan paikkaan, älä näytä tooltippiä

Kauko committed:
Karkeista löydettyjen urakoiden alueita

Kauko committed:
Näytä vetolaatikot latausvaiheessa, avaaminen piirtää spinnerin

Kauko committed:
Zoomaa kartta valittuihin urakoihin

Kauko committed:
Korjaa testit ely nimen muutoksen jälkeen

Kauko committed:
Merge branch 'develop' into HAR-2512_useamman_urakan_valinta_tilannekuvassa

Kauko committed:
Merge branch 'develop' into HAR-2512_useamman_urakan_valinta_tilannekuvassa

Kauko committed:
HAR-2638: korjaa valittujen alueiden esivalinta

Kauko committed:
Vahvista urakoiden rajoja

Kauko committed:
Palauta ylläpidon asiat vaikkei urakoita olisikaan valittuna

Kauko committed:
Hae ylläpidon juttuja vain urakoista, joihin on oikeus

Ylläpidon asiat voidaan hakea myös, kun urakka-lista on tyhjä.
Tällaisissa tapauksissa pitää vaan tarkistaa, että käyttäjällä
on varmasti yleinen oikeus tilannekuvaan.

Kauko committed:
Siirrä +koko-suomi-extent+ kartta.muuttujat namespaceen

Kauko committed:
Zoomaa koko Suomeen, kun mitään ei ole valittuna

Vaihdettiin zoomaus myös käyttämään add-watchia, eikä
run-blokkia.

Kauko committed:
Palauta tilannekuvassakin tarkastuksen laadunalitustieto

Kauko committed:
Näytä värillinen karttaselite reitillisille tarkastuksille

Kauko committed:
Merge branch 'develop' into HAR-2512_useamman_urakan_valinta_tilannekuvassa

Kauko committed:
Säilytä alueiden valinta tilannekuvassa kun näkymää vaihtaa

Kauko committed:
Muuta aluevalikon otsikko, kun ei ole näytettäviä alueita

Kauko committed:
Kunnioita murupolun tilaa ja edellisiä valintoja urakkaa valittaessa

Jos urakka on valittu murupolun kautta, ja kyseinen urakka on se,
valitaan urakka. Jos murupolusta on valittu vain HY, valitaan
kaikki HY:hyn liittyvät urakat. Vanhat valinnat kuitenkin
otetaan huomioon myös.

Kauko committed:
Näytä spinneri kun ensimmäisiä alueita ladataan näkymään tultaessa

Kauko committed:
Ota paremmin huomioon murupolun tila aluesuodattimen vanhan ja uuden tilan yhdistämisessä

Kauko committed:
Näytä hakuspinneri vain kun hakutulos on nil, ja suodattimet ovat tyhjät. Eli alussa

Kauko committed:
Merge branch 'develop' into HAR-2512_useamman_urakan_valinta_tilannekuvassa

Kauko committed:
HAR-2661: Työkoneen selitteet takaisin näkymään

Pitikin käyttää coll? eikä seq?. Lisäsin
viivojen-varit-leveimmasta-kapeimpaan funktioon ominaisuuden,
että jos sille antaa vaan yhden värin, niin se osaa palauttaa
sen suoraan. Siinä sitten epähuomiossa käytin väärää funktiota
tunnistamaan, että onko kyseessä collection vai ei.

Kauko committed:
Merge branch 'develop' into HAR-2512_useamman_urakan_valinta_tilannekuvassa

Kauko committed:
Poista reaktiosta<! :arvo-alussa

Kauko committed:
Poista tyhjä css sääntö

Kauko committed:
Katselmointimuutoksia

Kauko committed:
Siirrä koko-suomi-extent kartta/apurit namespaceen

Kauko committed:
Lisää kommentteja

Kauko committed:
Merge branch 'develop' into HAR-2512_useamman_urakan_valinta_tilannekuvassa

Kauko committed:
Uusiin raporttitesteihin oikea nimi pop hy:lle
